platform/posix/transport/src/transport_reconnect_posix.c

```diff
@@ -20,50 +20,60 @@
  */
 
 /**
- * @file reconnect_posix.c
+ * @file transport_reconnect_posix.c
  * @brief Implementation of the backoff logic when connection fails to the server fails.
  */
 
 /* Standard includes. */
 #include <unistd.h>
 #include <stdlib.h>
 #include <time.h>
-#include "reconnect.h"
+#include "transport_reconnect.h"
 
 /*-----------------------------------------------------------*/
 
 bool Transport_ReconnectBackoffAndSleep( TransportReconnectParams_t * pReconnectParams )
 {
     bool status = false;
-    uint32_t jitter = 0;
+    uint32_t nextBackOffAndJitter = 0;
 
     /* If MAX_RECONNECT_ATTEMPTS is set to 0, try forever */
-    if( ( pReconnectParams->attemptsDone < MAX_RECONNECT_ATTEMPS ) ||
-        ( 0 == MAX_RECONNECT_ATTEMPS ) )
+    if( ( pReconnectParams->attemptsDone < MAX_RECONNECT_ATTEMPTS ) ||
+        ( 0 == MAX_RECONNECT_ATTEMPTS ) )
     {
+        /* Reduce backoff value if more than max backoff time value. */
+        if( pReconnectParams->nextBackOffSec > MAX_RECONNECT_BACKOFF_SECONDS )
+        {
+            pReconnectParams->nextBackOffSec = MAX_RECONNECT_BACKOFF_SECONDS;
+        }
+
         /*  Wait for timer to expire for the next reconnect */
-        ( void ) sleep( pReconnectParams->reconnectTimeoutSec );
+        ( void ) sleep( pReconnectParams->nextBackOffSec );
+
+        /* Increment backoff counts. */
+        pReconnectParams->attemptsDone++;
```

Comment:
we may not want to increment this if  0 == MAX_RECONNECT_ATTEMPTS

Reply:
Can you explain why?
We need to keep a track of the count for calculating the upper limit value for `rand()` call in `sleep = random_between( 0, min( cap, base * 2 ** num_attempts ))`   equation.

Reply:
Attempts are only needed when we are limiting number of attempts. I thought upper limit is MAX_RECONNECT_BACKOFF_SECONDS, why do you need attempts ? Every iteration do 
value += value or value = 2*value.  That formula can be adjusted.